cmd/tailscale/cli/serve_dev.go

```diff
@@ -134,38 +144,31 @@ func newServeDevCommand(e *serveEnv, subcmd serveMode) *ffcli.Command {
 	}
 }
 
-// runServeCombined is the entry point for the "tailscale {serve,funnel}" commands.
-func (e *serveEnv) runServeCombined(subcmd serveMode) execFunc {
-	e.subcmd = subcmd
-
-	return func(ctx context.Context, args []string) error {
-		if len(args) == 0 {
-			return flag.ErrHelp
-		}
-
-		funnel := subcmd == funnel
-
+func (e *serveEnv) validateArgs(subcmd serveMode, args []string) error {
+	switch len(args) {
+	case 0:
+		return flag.ErrHelp
+	case 1, 2:
 		err := checkLegacyServeInvocation(subcmd, args)
```

Comment:
This error is not being used, let's make the function return a boolean, and rename to reflect: `isLegacyInvocation`